tests/DummyClient.py

```diff
@@ -0,0 +1,436 @@
+        base_ds : int
+            Hash / dataset ID of the base dataset or `None` if this is a root dataset.
+        types : list of str
+            State type name(s) of this state and its inner state(s).
+        root : bool
+            `True` if this is a root dataset (default `False`).
+        dump : bool
+            Tells the broker if the state should be dumped to file or not.
+        timestamp : str
+            Overwrite the timestamp attached to this dataset. Should have the format `DummyClient.TIMESTAMP_FORMAT`.
+        ds_id : int
+            Manually set the hash of this dataset instead of letting CoMeT compute it.
+
+        Raises
+        ------
+        :class:`ManagerError`
+            If there was an internal error in the dataset management.
+        :class:`BrokerError`
+            If there was an error in registering stuff with the broker.
+        :class:`ConnectionError`
+            If the broker can't be reached.
+        """
+        if not isinstance(state, int):
+            raise ManagerError(
+                "state needs to be a hash (int) (is `{}`).".format(type(state).__name__)
+            )
+        if not self.start_state:
+            raise ManagerError("Start has to be registered before anything else")
+
+        ds = {"is_root": root, "state": state, "types": types}
+        if base_ds is not None:
+            ds["base_dset"] = base_ds
+
+        if ds_id is None:
+            ds_id = self._make_hash(ds)
+
+        request = {"hash": ds_id, "ds": ds, "dump": dump}
+        if timestamp:
+            request["time"] = timestamp
+        self._send(REGISTER_DATASET, request)
+
+        self.datasets[ds_id] = ds
+
+        return ds_id
+
+    @stopwatch
+    def update_datasets(self, ds_id, timestamp=0, roots=[]):
+        """Query for an update on the dataset.
+
+        Requests all nodes that were after the given timestamp.
+
+        Parameters
+        ----------
+        ds_id : int
+            Hash for the dataset you wish to update.
+        timestamp : str
+            Request all nodes that were after the given timestamp.
+            If 0, return all nodes.
+        roots : list of int
+            Hash for the given roots that are included in the returned update.
+            If the root of the given dataset is not among them, all datasets with the same root as the given dataset are returned.
+        """
+        if not isinstance(ds_id, int):
+            raise ManagerError(
+                "ds_id needs to be a hash (int) (is `{}`).".format(type(state).__name__)
+            )
+
+        if not self.start_state:
+            raise ManagerError("Start has to be registered before anything else")
+
+        request = {"ds_id": ds_id, "ts": timestamp, "roots": roots}
+
+        response = self._send(UPDATE_DATASETS, request)
+
+        return response
+
+    @staticmethod
+    def _make_hash(data):
+        return hash(json.dumps(data, sort_keys=True))
```

Comment:
The hash just changed to 128bit MurmurHash, e.g.:
```
    >>> import mmh3
    >>> h = mmh3.hash128(b"This is a long and random string.", seed=1420)
```

Reply:
So far, the ints it makes are too big and json.loads rejects them! Trying to figure it out...

Reply:
Ref: https://github.com/pandas-dev/pandas/issues/14530

Reply:
I think we have to serialize a single hash into: `"{:016x}{:016x}"`